### PR TITLE
add Amazon SES email adapter

### PR DESCRIPTION
**Pull request by Schparky**

### CODE REVIEW

File: application/notifications/ses.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	"github.com/silinternational/wecarry-api/domain"
+)
+
+// SES sends email using Amazon Simple Email Service (SES)
+type SES struct {
+}
+
+// Send a message
+func (s *SES) Send(msg Message) error {
+	apiKey := domain.Env.SendGridAPIKey
```
Comment (fillup) [inline application/notifications/ses.go]: Is this right?

File: application/domain/domain.go [hunk 2]
```diff
@@ -142,12 +144,14 @@ func readEnv() {
...
 	Env.AppName = envy.Get("APP_NAME", "WeCarry")
 	Env.AuthCallbackURL = envy.Get("AUTH_CALLBACK_URL", "")
-	Env.AwsS3Region = envy.Get("AWS_REGION", "")
+	Env.AwsRegion = envy.Get("AWS_REGION", "")
 	Env.AwsS3Endpoint = envy.Get("AWS_S3_ENDPOINT", "")
 	Env.AwsS3DisableSSL, _ = strconv.ParseBool(envy.Get("AWS_S3_DISABLE_SSL", "false"))
 	Env.AwsS3Bucket = envy.Get("AWS_S3_BUCKET", "")
 	Env.AwsS3AccessKeyID = envy.Get("AWS_S3_ACCESS_KEY_ID", "")
 	Env.AwsS3SecretAccessKey = envy.Get("AWS_S3_SECRET_ACCESS_KEY", "")
+	Env.AwsSESAccessKeyID = envy.Get("AWS_SES_ACCESS_KEY_ID", Env.AwsS3AccessKeyID)
```
Comment (Baggerone) [inline application/domain/domain.go]:
```suggestion
	Env.AwsSESAccessKeyID = envy.Get("AWS_SES_ACCESS_KEY_ID", Env.AwsSESAccessKeyID)
```

File: application/domain/domain.go [hunk 3]
```diff
@@ -142,12 +144,14 @@ func readEnv() {
...
+	Env.AwsSESSecretAccessKey = envy.Get("AWS_SES_SECRET_ACCESS_KEY", Env.AwsS3SecretAccessKey)
```
Comment (Baggerone) [inline application/domain/domain.go]:
```suggestion
	Env.AwsSESSecretAccessKey = envy.Get("AWS_SES_SECRET_ACCESS_KEY", Env.AwsSES3SecretAccessKey)
```

File: application/notifications/ses.go (hunk 1 above)
Comment (Schparky) [reply application/notifications/ses.go]: Nope, thanks.

File: application/domain/domain.go (hunk 2 above)
Comment (Schparky) [reply application/domain/domain.go]: My intent was to default to use the S3 key.

File: application/aws/aws.go [hunk 4]
```diff
@@ -159,3 +161,69 @@ func CreateS3Bucket() error {
...
+		},
+		Source: aws.String(from),
+	}
+
+	result, err := svc.SendEmail(input)
+	if err != nil {
+		return fmt.Errorf("SendEmail failed using SES, %s", err)
+	}
+
+	domain.Logger.Printf("Message sent, message ID: %s", *result.MessageId)
```
Comment (Baggerone) [inline application/aws/aws.go]: Is it worth specifying that it's an AWS SES message that was sent?

**Pull request merged**